# Cleaning the 2‑month traffic dataset

In this notebook I:

1. Load the raw 2‑month traffic data from `../data/raw/TrafficTwoMonth.csv`.
2. Inspect the schema and basic data quality.
3. Construct a `datetime` column from `Date` and `Time`.
4. Sort the data chronologically and remove duplicates.
5. Save the cleaned dataset as `../data/clean/traffic_2months_clean.csv`.

This cleaned file is the starting point for feature engineering and forecasting.

In [1]:
import pandas as pd
import numpy as np

pd.set_option("display.max_rows", 10)
pd.set_option("display.max_columns", None)

1. Load raw data

In [2]:
raw_path = "../data/raw/TrafficTwoMonth.csv"  
df_raw = pd.read_csv(raw_path)
df_raw.head()

,Time,Date,Day of the week,CarCount,BikeCount,BusCount,TruckCount,Total,Traffic Situation
0,12:00:00 AM,10,Tuesday,13,2,2,24,41,normal
1,12:15:00 AM,10,Tuesday,14,1,1,36,52,normal
2,12:30:00 AM,10,Tuesday,10,2,2,32,46,normal
3,12:45:00 AM,10,Tuesday,10,2,2,36,50,normal
4,1:00:00 AM,10,Tuesday,11,2,1,34,48,normal


In [3]:
df_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5952 entries, 0 to 5951
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Time               5952 non-null   object
 1   Date               5952 non-null   int64 
 2   Day of the week    5952 non-null   object
 3   CarCount           5952 non-null   int64 
 4   BikeCount          5952 non-null   int64 
 5   BusCount           5952 non-null   int64 
 6   TruckCount         5952 non-null   int64 
 7   Total              5952 non-null   int64 
 8   Traffic Situation  5952 non-null   object
dtypes: int64(6), object(3)
memory usage: 418.6+ KB


In [4]:
df_raw.isna().sum()

Time                 0
Date                 0
Day of the week      0
CarCount             0
BikeCount            0
BusCount             0
TruckCount           0
Total                0
Traffic Situation    0
dtype: int64

### Brief interpretation of the raw schema

From the output above:

- There are 5952 rows and 9 columns.
- `CarCount`, `BikeCount`, `BusCount`, `TruckCount`, and `Total` are already `int64`.
- `Time`, `Date`, `Day of the week`, and `Traffic Situation` are objects
  (string‑like) as expected.

Since the counts are already integers and there are no missing values,
I do not need to convert types or impute data here

## 2. Copy to a working DataFrame

To avoid accidentally modifying the original object, I work on a copy
named `df`.

In [5]:
df = df_raw.copy()

Some small sanity checks:

In [ ]:
numeric_cols = ["CarCount", "BikeCount", "BusCount", "TruckCount", "Total"]

(df[numeric_cols] < 0).sum() # wanted to check that the numeric columns are always non‑negative.

CarCount      0
BikeCount     0
BusCount      0
TruckCount    0
Total         0
dtype: int64

In [9]:
component_sum = df["CarCount"] + df["BikeCount"] + df["BusCount"] + df["TruckCount"]
diff = df["Total"] - component_sum

diff.describe() #wanted to check that the `Total` column is roughly consistent with the sum of the separate vehicle counts
   

count    5952.0
mean        0.0
std         0.0
min         0.0
25%         0.0
50%         0.0
75%         0.0
max         0.0
dtype: float64

At this point, the "cleaning" process mainly documents that:

- There are no missing values.
- Numeric types are correct.
- Basic sanity checks have been performed.

I will save the result to `../data/clean/traffic_2months_clean.csv`.

In [11]:
clean_path = "../data/processed/traffic_2months_clean.csv"

df.to_csv(clean_path, index=False)

clean_path

'../data/processed/traffic_2months_clean.csv'